File: splearn-java/src/main/java/io/wisoft/splearn/application/required/MemberRepository.java
package io.wisoft.splearn.application.required;

import io.wisoft.splearn.domain.Email;
import io.wisoft.splearn.domain.Member;
import org.springframework.data.repository.Repository;

import java.util.Optional;

/**
 * 회원 정보를 저장하거나 조회한다
 */
public interface MemberRepository extends Repository<Member, Long> {
    Member save(Member member);

    Optional<Member> findByEmail(Email email);
}

File: splearn-typescript/src/main/application/member/required/member.repository.ts
import { Email } from "@src/main/domain/shared/email";
import { Member } from "@src/main/domain/member/member";
import { Profile } from "@src/main/domain/member/profile";

/**
 * 회원 정보를 저장하거나 조회한다
 */
export interface MemberRepository {
  save(member: Member): Promise<Member>;

  findByEmail(email: Email): Promise<Member | null>;

  findById(memberId: number): Promise<Member | null>;

  findByProfile(profile: Profile): Promise<Member | null>;
}